omod/src/main/java/org/openmrs/module/smartonfhir/web/filter/SmartSessionLogoutFilter.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.smartonfhir.web.filter;
+
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import java.io.IOException;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.keycloak.adapters.AdapterDeploymentContext;
+import org.keycloak.adapters.KeycloakDeploymentBuilder;
+import org.keycloak.adapters.spi.HttpFacade;
+import org.keycloak.adapters.springsecurity.facade.SimpleHttpFacade;
+import org.openmrs.module.smartonfhir.util.KeycloakConfigHolder;
+
+@Slf4j
+public class SmartSessionLogoutFilter implements Filter {
+	
+	private volatile String logoutUrl = null;
+	
+	private volatile AdapterDeploymentContext adapterDeploymentContext = null;
+	
+	@Override
+	public void init(FilterConfig filterConfig) {
+		adapterDeploymentContext = new AdapterDeploymentContext(
+		        KeycloakDeploymentBuilder.build(KeycloakConfigHolder.getKeycloakConfig()));
+	}
```

Comment:
I actually think we can do what we need to do like this:

```
	@Override
	public void init(FilterConfig filterConfig) {
		logoutUrl = KeycloakDeploymentBuilder.build(KeycloakConfigHolder.getKeycloakConfig()).getLogoutUrl();
	}
```

This eliminates the need for the `adapterDeploymentContext` at all as well as the `volatile` keyword and the `getLogoutUrl()` method.

Reply:
@ibacher This method is way more easier than the last one. I think `KeycloakDeploymentBuilder.build(KeycloakConfigHolder.getKeycloakConfig()).getLogoutUrl();` return `KeycloakUriBuilder`object, we have to construct the complete logout URL, sometime like this:
```
logoutUrl = keycloakUriBuilder.getScheme() + "://" + keycloakUriBuilder.getHost() + ":"
		        + keycloakUriBuilder.getPort() + keycloakUriBuilder.getPath();
```

Thought?

Reply:
You're right... I missed that... maybe just call `toTemplate()` on the `KeycloakUriBuilder` and see what that gets us (I don't imagine the logout URL takes too many parameters)

Reply:
Yes, we are getting the same URL. Thanks